src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -0,0 +1,231 @@
+    });
+    if (
+      selectedItems[multiSelectId] !== undefined &&
+      childIds.every((childItem) =>
+        selectedItems[multiSelectId].items.includes(childItem)
+      )
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  const isIndeterminate = (
+    multiSelectId: string,
+    item: MultiSelectItem
+  ): boolean => {
+    let childIds = [];
+    item.children.map((childItem) => {
+      childIds.push(childItem.id);
+    });
+    if (
+      selectedItems[multiSelectId] !== undefined &&
+      childIds.some((childItem) =>
+        selectedItems[multiSelectId].items.includes(childItem)
+      ) &&
+      !isAllChecked(multiSelectId, item)
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  return (
+    <Box
+      id={id}
+      ref={ref}
+      __css={styles}
+      {...(isOpen && { "aria-modal": true })}
+    >
+      <FocusLock isDisabled={!isOpen}>
+        <MultiSelectMenuButton
+          multiSelectId={id}
+          label={label}
+          isOpen={isOpen}
+          selectedItems={selectedItems}
+          onMenuToggle={() => {
+            setIsOpen(!isOpen);
+          }}
+          onClear={onClear}
+        />
+        <Box
+          role="dialog"
+          __css={styles.menuContainer}
+          {...(!isOpen && { display: "none" })}
+        >
+          <UnorderedList
+            styleType="none"
+            marginInlineStart="0"
+            __css={styles.menu}
+          >
+            {isOpen &&
+              items.map((item: MultiSelectItem) => (
+                <ListItem key={item.id} py={1}>
+                  {item.children ? (
+                    <>
+                      <Checkbox
+                        id={item.id}
+                        labelText={item.name}
+                        showLabel={true}
+                        name={item.name}
+                        // If onMixedStateChange is not passed as a prop, handle
+                        // the parent checkbox as a regular checkbox using onChange
+                        {...(onMixedStateChange !== undefined
+                          ? {
+                              isChecked: isAllChecked(id, item) || false,
+                              isIndeterminate:
+                                isIndeterminate(id, item) || false,
+                              onChange: onMixedStateChange,
+                            }
+                          : {
+                              isChecked: isChecked(id, item.id) || false,
+                              onChange: onChange,
+                            })}
+                      />
+                      <UnorderedList
+                        styleType="none"
+                        marginInlineStart="0"
+                        __css={styles.menuChildren}
+                      >
+                        {item.children.map((childItem) => {
+                          return (
+                            <ListItem key={childItem.id} py={1}>
```

Comment:
Use the NYPL design token value rather than the number value. In this case `py="xxs"` instead of `p={1}`.

Reply:
fixed